asmcomp/selectgen.ml

```diff
@@ -896,23 +891,21 @@ method private emit_sequence (env:environment) exp =
   let r = s#emit_expr env exp in
   (r, s)
 
-method private bind_let (env:environment) v r1 =
-  if all_regs_anonymous r1 then begin
+method private bind_let (env:environment) kind v r1 =
+  (* REVIEW why the kind = Immutable condition in the anonymous path? *)
+  if kind = Immutable && all_regs_anonymous r1 then begin
     name_regs v r1;
     env_add v r1 env
   end else begin
-    let rv = Reg.createv_like r1 in
+    let rv = match kind with
+      | Immutable -> Reg.createv_like r1
+      | Mutable mty -> Reg.createv mty
+    in
     name_regs v rv;
     self#insert_moves env r1 rv;
     env_add v rv env
   end
 
-method private bind_let_mut (env:environment) v k r1 =
-  let rv = Reg.createv k in
-  name_regs v rv;
-  self#insert_moves env r1 rv;
-  env_add ~mut:Mutable v rv env
```

Comment:
The additional `~mut:Mutable` argument is missing in the new `bind_let`

Reply:
Oops... fixed.